src/flows/components/VersionSidebar.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {FC, createRef, useContext, RefObject} from 'react'
+import {useHistory, useParams} from 'react-router-dom'
+import {useSelector} from 'react-redux'
+
+// Components
+import {
+  AlignItems,
+  Button,
+  ComponentColor,
+  ComponentSize,
+  DapperScrollbars,
+  FlexBox,
+  FlexDirection,
+  Icon,
+  IconFont,
+  InfluxColors,
+  List,
+  Popover,
+  PopoverInteraction,
+} from '@influxdata/clockface'
+
+// Context
+import {FlowContext} from 'src/flows/context/flow.current'
+import {VersionPublishContext} from 'src/flows/context/version.publish'
+
+// Utils & Actions
+import {event} from 'src/cloud/utils/reporting'
+import {getOrg} from 'src/organizations/selectors'
+import {
+  patchNotebook,
+  postNotebook,
+  VersionHistory,
+} from 'src/client/notebooksRoutes'
+import {getAllAPI} from 'src/flows/context/api'
+import {incrementCloneName} from 'src/utils/naming'
+import {serialize} from 'src/flows/context/flow.list'
+
+// Constants
+import './Sidebar.scss'
+import {PROJECT_NAME, PROJECT_NAME_PLURAL} from 'src/flows'
+import {Flow} from 'src/types'
+
+interface Props {
+  version: VersionHistory | Flow
+}
+
+const VersionSidebarListItem: FC<Props> = ({version}) => {
+  const triggerRef: RefObject<HTMLButtonElement> = createRef()
+  const history = useHistory()
+  const {notebookID, id} = useParams<{notebookID: string; id: string}>()
```

Comment:
can we explore notebooks without having a current notebook context? if not, we should be decoupling the view from the route as much as possible and using the notebook id from the flow context. i think most of your code reflects this already, so imma ask that you remove  this line, but i might be missing something? seems like we have two sources of "current notebook" and "current version" being used here